### PR TITLE
Add OpenSSL code from jdk12 

### PR DESCRIPTION
**Pull request by groeges**

### CODE REVIEW

**Comment by groeges** [comment]
jenkins test sanity.functional xlinux

**Comment by groeges** [comment]
jenkins compile xlinux

**Comment by pshipton** [comment]
jenkins test sanity plinux

**Comment by pshipton** [comment]
jenkins test sanity xlinux

**Comment by pshipton** [comment]
Created https://github.com/eclipse/openj9/issues/5037
Also realized test sanity isn't helpful, because OpenJ9 needs changes to enable openssl first.

**Comment by pshipton** [comment]
Started plinux testing from https://github.com/eclipse/openj9/pull/5038

**Comment by pshipton** [comment]
`--with-openssl=system` doesn't work
```
12:40:19  checking for OPENSSL... checking for OPENSSL... no
12:40:19  configure: error: Unable to find openssl 1.0.2(and above) installed on System. Please use other options for '--with-openssl'
```

**Comment by pshipton** [comment]
CryptoTest has passed when using `--with-openssl=fetched`

**Comment by groeges** [comment]
> `--with-openssl=system` doesn't work
> 
> ```
> 12:40:19  checking for OPENSSL... checking for OPENSSL... no
> 12:40:19  configure: error: Unable to find openssl 1.0.2(and above) installed on System. Please use other options for '--with-openssl'
> ```

@pshipton The custom-hook.m4 states:
```
      # We can use the system installed openssl only when it is package installed.
      # If not package installed, fail with an error message.
```
What OS were you running this on. If Linux, does the `pkg-config` command show that openssl is installed. Try running `pkg-config --print-errors openssl` to see whether it gets found or not.

**Comment by groeges** [comment]
@pshipton This also failed on one of my Linux boxes. 
In order to get it to work I had to also install `openssl-dev`

**Comment by pshipton** [comment]
> In order to get it to work I had to also install openssl-dev

Ok. OpenJ9 can continue to use `--with-openssl=fetched`

File: closed/src/java.base/share/native/libjncrypto/NativeCrypto_md.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+#ifndef NATIVECRYPTO_MD_H
+#define NATIVECRYPTO_MD_H
+
+void * load_crypto_library();
+void   unload_crypto_library(void *handle);
+void * find_crypto_symbol(void *handle, const char *symname);
+
+#endif
```
Comment (davehobbs) [inline closed/src/java.base/share/native/libjncrypto/NativeCrypto_md.h]: I added a /* NATIVECRYPTO_MD_H */ comment to the endif

Comment (groeges) [reply closed/src/java.base/share/native/libjncrypto/NativeCrypto_md.h]: Done

**Comment by groeges** [comment]
jenkins test sanity xlinux

**Comment by groeges** [comment]
jenkins build xlinux

**Comment by andrew-m-leonard** [comment]
jenkins test sanity xlinux

**Comment by groeges** [comment]
@andrew-m-leonard Those jenkins jobs don't work on jdk (13) repo yet!!

**Comment by pshipton** [comment]
Also note that OpenJ9 doesn't enable building openssl for jdk next. Testing for this needs to be done via eclipse/openj9#5038. I tested plinux previously, I can run it again if desired.

Only PR testing for xlinux and plinux are supported for jdk next atm. I've opened eclipse/openj9#5045 to address this.

**Comment by andrew-m-leonard** [comment]
thanks Peter

**Comment by groeges** [comment]
@pshipton @andrew-m-leonard Are we OK to get this merged now

**Comment by andrew-m-leonard** [comment]
@pshipton ?

**Comment by andrew-m-leonard** [comment]
From what tests we can see looks ok

**Pull request merged**